addons/binding/org.openhab.binding.vera/src/main/java/org/openhab/binding/vera/internal/converter/VeraDeviceStateConverter.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.vera.internal.converter;
+
+import static org.openhab.binding.vera.VeraBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.vera.controller.json.Device;
+import org.slf4j.Logger;
+
+/**
+ * The {@link VeraDeviceStateConverter} is responsible for converting Vera device level to openHAB states
+ *
+ * @author Dmitriy Ponomarev
+ */
+public class VeraDeviceStateConverter {
+    public static State toState(Device device, Channel channel, Logger logger) {
+
+        String channelType = channel.getUID().getId().split("-")[0];
+        if (BATTERY_CHANNEL.equals(channelType)) {
+            return getMultilevelState(device.getBatterylevel());
+        }
+        if (SENSOR_METER_KWH_CHANNEL.equals(channelType)) {
+            return getMultilevelState(device.getKwh());
+        }
+        if (SENSOR_METER_W_CHANNEL.equals(channelType)) {
+            return getMultilevelState(device.getWatts());
+        }
+
+        int subcategory = Integer.parseInt(device.getSubcategory());
+        switch (device.getCategoryType()) {
+            case Controller:
+            case Interface:
+                break;
+            case DimmableLight:
+                switch (subcategory) {
+                    case 1:
+                    case 2:
+                    case 3:
+                        return getPercentState(device.getLevel());
+                    case 4:
+                        return getPercentState(device.getLevel());// TODO getColorState(device.light);
+                }
+                break;
+            case Switch:
+                return getBinaryState(device.getStatus());
+            case SecuritySensor:
+                switch (subcategory) {
+                    case 1:
+                        return getDoorlockState("1".equals(device.getTripped()) ? "0" : "1");
+                    default:
+                        return getBinaryState(device.getTripped());
+                }
+            case HVAC: // TODO
+                logger.warn("TODO: {}, {}", device, device.getCategoryType());

```

Comment:
Can you change it to `debug` instead? I wouldn't like a log file full of warnings just because devices aren't yet supported by a binding. It would also help to make a note about this in the docs.